netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -176,20 +222,109 @@ public static Status statusFromThrowable(Throwable t) {
     return s;
   }
 
+  @VisibleForTesting
+  static boolean isEpollAvailable() {
+    try {
+      boolean available = (Boolean)
+          Class
+              .forName("io.netty.channel.epoll.Epoll")
+              .getDeclaredMethod("isAvailable")
+              .invoke(null);
+      if (!available) {
+        logger.log(Level.FINE, "Epoll is not available", getEpollUnavailabilityCause());
+      }
+      return available;
+    } catch (Exception e) {
```

Comment:
ClassNotFound should probably be special-cased, since that is quite normal (although it would still be fair to log it at FINE). Otherwise this exception should be logged, probably at WARNING.